weblate/trans/models/comment.py

```diff
@@ -17,25 +18,38 @@
 from weblate.utils.state import STATE_FUZZY
 
 if TYPE_CHECKING:
-    from weblate.auth.models import AuthenticatedHttpRequest, User
+    from datetime import datetime
+
+    from weblate.auth.models import AuthenticatedHttpRequest, Unit, User
 
 
 class CommentManager(models.Manager):
-    def add(self, request: AuthenticatedHttpRequest, unit, text, scope) -> None:
+    def add(
+        self,
+        request: AuthenticatedHttpRequest,
+        unit: Unit,
+        text: str,
+        scope: str,
+        user: User,
+        timestamp: datetime | None = None,
```

Comment:
I'd prefer to keep the existing semantics to extract user from the request if not set. Also making new args kwargs only will make the code easier to read with many parameters:

```suggestion
        *,
        user: User | None = None,
        timestamp: datetime | None = None,
```